app/models/story.rb

```diff
@@ -1008,4 +1013,15 @@ def valid_canonical_uri?(url)
   def canonical_target(parsed)
     parsed.at_css("link[rel='canonical']").attributes["href"].text
   end
+
+  def build_normalized_domain
+    begin
+      hostname = URI.parse(url).hostname.squish
```

Comment:
See `URL_RE` in this story - we don't use `URI.parse` much because it's not lenient enough. We'll need to use that like `set_domain` did.